modules/kernel/src/org/apache/axis2/deployment/DescriptionBuilder.java

```diff
@@ -691,15 +698,43 @@ protected String getValue(String in) {
     }
 
     protected String replaceSystemProperty(String text) {
+        String[] sysRefs = StringUtils.substringsBetween(text, SYS_PROPERTY_PLACEHOLDER_PREFIX, PLACEHOLDER_SUFFIX);
+        String[] envRefs = StringUtils.substringsBetween(text, ENV_VAR_PLACEHOLDER_PREFIX, PLACEHOLDER_SUFFIX);
+
+        // Resolves system property references ($sys{ref}) in an individual string.
+        if (sysRefs != null) {
+            for (String ref : sysRefs) {
+                String property = System.getProperty(ref);
+                if (StringUtils.isNotEmpty(property)) {
+                    text = text.replaceAll(Pattern.quote(SYS_PROPERTY_PLACEHOLDER_PREFIX + ref + PLACEHOLDER_SUFFIX), property);
+                } else {
+                    log.warn("System property is not available for " + ref);
```

Comment:
If it is not available for one property, is there any use in iterating though the list and returning ?. Shouldn't we return empty string. Also this has to be an **error** log ?

Reply:
Fixed in 7bc767446d2e485f2bfd7f274684d8137f323ef4